portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +13,95 @@
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    
+    // instance vairables 
+    ArrayList<String> messages;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Fran you did it you solved!!");
+        Query query = new Query("comment").addSort("timestamp", SortDirection.DESCENDING);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        // reset messages in order to always showcase all the comments since start of time in my webpage 
+        messages = new ArrayList<String>();
+        
+        // TODO: make a comment type 
+        for (Entity entity : results.asIterable()) {
+            long id = entity.getKey().getId();
+            String username = (String) entity.getProperty("username");
+            String text = (String) entity.getProperty("text");
+            long timestamp = (long) entity.getProperty("timestamp");
+
+            String completeComment = username + " said " + text + " at " + timestamp; 
+            if(messages != null){
+                messages.add(completeComment);
+            }
+        }
+
+    // only show the messages and print json if messages not null
+    if(messages.size() != 0){
+        // converting the message array list to a json 
+        String jsonString = convertToJsonUsingGson(messages);
+
+        // send the json as the response  
+        response.setContentType("application/json;");
+        response.getWriter().println(jsonString);
+    }
+  }
+
+   @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // data store & entity creation 
+    Entity taskEntity = new Entity("comment");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  
+    // recieve the comments from response.getParamaters 
+    String userName = getParamatersDataMassaged(request, "username");
+    String specificComment = getParamatersDataMassaged(request, "comment");
+    long timestamp = System.currentTimeMillis();
+
+    // update entity properties to be of the 
```

Comment:
Looks like an incomplete comment

Reply:
Ah yea I think I meant that i was updating the entity properties to assign the parameters the user passed in. I edited the comment to explain that